### DISCUSSION TITLE
How to publish incoming call via MQTT module?

### DISCUSSION THREAD

**Discussion by duffy6** [baresip/baresip#3246] [Q&A]
Hi guys,
ich have baresip running as a container and it connects to my SIP Server (german Fritzbox router) and my MQTT Server:

`root@eb6b2b2da495 :~ /baresip# baresip -f /root/baresip
baresip v1.0.0 Copyright (C) 2010 - 2020 Alfred E. Heggestad et al.
Local network address: IPv4=eth0|172.17.0.3
ua: adding SIP CA: /etc/ssl/certs/ca-certificates.crt
aucodec: PCMU/8000/1
aucodec: PCMA/8000/1
dl: mod: /usr/lib/baresip/modules/auconv.so (/usr/lib/baresip/modules/auconv.so:
cannot open shared object file: No such file or directory)
module auconv.so: No such file or directory
dl: mod: /usr/lib/baresip/modules/auresamp.so (/usr/lib/baresip/modules/auresamp
so: cannot open shared object file: No such file or directory)
module auresamp.so: No such file or directory
ausrc: alsa
auplay: alsa
medianat: stun
medianat: turn
medianat: ice
Populated 1 account
Populated 3 contacts
mqtt: connecting to broker at 192.168.133.59:1883 as baresip01 topic baresip/01
mqtt: Publishing on /baresip/01/event, subscribing to /baresip/01/command/+
mqtt: module loaded
dl: mod: /usr/lib/baresip/modules/ctrl_dbus.so (/usr/lib/baresip/modules/ctrl dbu
S.so: cannot open shared object file: No such file or directory)
module ctrl_dbus.so: No such file or directory
dl: mod: /usr/lib/baresip/modules/netroam.so (/usr/lib/baresip/modules/netroam.so
: cannot open shared object file: No such file or directory)
module netroam.so: No such file or directory
Populated 2 audio codecs
Populated 0 audio filters
Populated 0 video codecs
Populated 0 video filters
baresip is ready.
mqtt: connected to broker at 192.168.133.59:1883
mqtt: subscribed to pattern '/baresip/01/command/+'
iobroker@fritz.box: {0/UDP/v4} 200 OK () [1 binding]
All 1 useragent registered successfully! (37 ms )
`

But I do not receive a MQTT Message for an incoming call.
The only thing I see in my MQTT topic "/baresip/01/event" is this:

`{"type":"REGISTER_OK","class":"register","accountaor":"sip:iobroker@fritz.box","param":"200 OK"}`

How can I publish an incoming call and the incoming number with the mqtt.so module?

**Comment by sreimers** [comment]
> baresip v1.0.0

is quite old and not supported anymore, can you try latest release please. We have a official container here: https://github.com/baresip/docker

**Reply by duffy6** [reply]
Thanks! I used the latest Dockerfile.
I still do not see any mqtt messages for incoming calls.

This is what I get after starting baresip:
```eth0: 172.17.0.3
ua: adding SIP CA file: /etc/ssl/certs/ca-certificates.crt
ua: adding SIP CA path: /etc/ssl/certs
aucodec: PCMU/8000/1
aucodec: PCMA/8000/1
aufilt: auconv
aufilt: auresamp
ausrc: alsa
auplay: alsa
medianat: stun
medianat: turn
medianat: ice
Populated 1 account
Populated 3 contacts
mqtt: connecting to broker at 192.168.133.59:1883 as baresip01 topic baresip/01
mqtt: Publishing on /baresip/01/event, subscribing to /baresip/01/command/+
mqtt: module loaded
Populated 2 audio codecs
Populated 2 audio filters
Populated 0 video codecs
Populated 0 video filters
baresip is ready.
mqtt: connected to broker at 192.168.133.59:1883
mqtt: subscribed to pattern '/baresip/01/command/+'
iobroker@fritz.box: (prio 0) {0/UDP/v4} 200 OK () [1 binding]
All 1 useragent registered successfully! (58 ms)```

Any recoomendation how I can get notifications?

**Reply by cspiel1** [reply]
I tested with main HEAD and a mosquitto broker with TLS. Subscribed with:
```
mosquitto_sub -h <brocker-host> -p <brocker-port> -u <user> -P <password> --capath /etc/ssl/certs -t "/baresip/+"
```
Got these mqtt messages:
```
{"class":"call","type":"CALL_REMOTE_SDP","accountaor":"[sip:A@10.1.0.2](mailto:sip%3AA@10.1.0.2)","direction":"incoming","peeruri":"sip:alice@office","localuri":"sip:10.1.0.2","remoteaudiodir":"sendrecv","remotevideodir":"sendrecv","audiodir":"sendrecv","videodir":"sendrecv","localaudiodir":"sendrecv","localvideodir":"sendrecv","param":"offer"}

{"class":"call","type":"CALL_INCOMING","accountaor":"[sip:A@10.1.0.2](mailto:sip%3AA@10.1.0.2)","direction":"incoming","peeruri":"sip:alice@office","localuri":"sip:10.1.0.2","id":"4fa05195be16de9d","remoteaudiodir":"sendrecv","remotevideodir":"sendrecv","audiodir":"sendrecv","videodir":"sendrecv","localaudiodir":"sendrecv","localvideodir":"sendrecv","param":"sip:alice@office"}

{"class":"call","type":"CALL_CLOSED","accountaor":"[sip:A@10.1.0.2](mailto:sip%3AA@10.1.0.2)","direction":"incoming","peeruri":"sip:alice@office","localuri":"sip:10.1.0.2","id":"4fa05195be16de9d","remoteaudiodir":"sendrecv","remotevideodir":"sendrecv","audiodir":"sendrecv","videodir":"sendrecv","localaudiodir":"sendrecv","localvideodir":"sendrecv","param":"Connection reset by peer [104]"}
```
You could check your broker setup and the subscribe topic.

Short: The topic should be "/baresip/+"
